src/js/common/schemaform/reducers/index.js

```diff
@@ -21,12 +21,19 @@ function recalculateSchemaAndData(initialState) {
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = _.mapValues('data', state);
```

Comment:
Mmm, yeah I didn't want to do that to `updateSchema` because for that, the idea was to actually update the `schema` / `uiSchema` objects inside each page, so it didn't make as much sense.